airflow/contrib/hooks/bigquery_hook.py

```diff
@@ -486,6 +487,11 @@ def run_load(self,
                 'writeDisposition': write_disposition,
             }
         }
+        
+        # if it is a partitioned table ($ is in the table name) add partition load option
+        if '$' in destination_project_dataset_table:
```

Comment:
I'd consider a regex here rather than just a check on the $ character

Reply:
I used $ as it is the partition decorator separator in BQ and won't change. 
Also BQ doe not allow special characters in the table or dataset name (except the partition separator).